accepted/0000-payg-cloud-integration.md

```diff
@@ -0,0 +1,209 @@
+
+Cloud RMT IP needs to be added to `/ets/hosts` on uyuni server. IP and hostname are returned by the data extraction script.
+
+Cloud RMT https certificate is also returned by the data extraction tool and on uyuni server we need to:
+  - add the certificate to folder `/etc/pki/trust/anchors/<label>`
+  - run command `update-ca-certificates`
+
+### Implementation steps - Register new PAYG repositories
+
+After we get all repositories information from the PAYG instance, for each repository we will need to:
+
+#### Find internal suse Scc repository ID
+
+Repositories are identified by its URL. We can extract repository endpoint from PAYG machine and find the corresponding repository in Uyuni database. Note that the cloud RMT server have a url prefix `/repo/` which we need to remove before compare.
+
+Example:
+- URL on PAYG instance: `https:\\host/repo/SUSE/Products/SLE-Module-Basesystem/15-SP2/x86_64/product/`
+- URL uyuni database (suseSccRepository table): `https://updates.suse.com/SUSE/Products/SLE-Module-Basesystem/15-SP2/x86_64/product/`
+
+Query to be executed: `select id from susesccrepository where url like '%SUSE/Products/SLE-Module-Basesystem/15-SP2/x86_64/product/';`
+
+The query above will return the internal suse scc repository ID, to be used in next step.
+
+#### Insert repository authentication data
+
+- If not exists, add a new entry to table `susecredentials`, with the new authentication type `cloudrmt`, containing the repository basic authentication credentials.
+- Add a new entry to table `susesccrepositoryauth` for each repository, where:
+  - repo_id: the one obtained in previous step
+  - credentials_id: the one obtain from `susecredentials`
+  - source_id: null
+  - auth_type: `cloudrmt`
+  - auth: authentication header obtained from the PAYG instance for the repository
+
+We also need to save the repository URL, since we will not be able to use the existing SCC repository url.
+If this information will be saved in a new table field of `susesccrepositoryauth` table or in a new table is a implementation detail.
+
+
+### Implementation steps - Add product with setup wizard
+
+When a product is added in the setup wizard table `rhncontentsource` is populated according to the authentication mechanism defined in table `susesccrepositoryauth` using the repository url defined in table `susesccrepository`.
+
+We need to adapt this feature to consider the new `cloudrmt` authentication mechanism and copy the basic authentication and headers:
+  - basic authentication can be defined on repo URL (`https://user:pass@host/repo/url/`)
+  - authentication header location is an implementation detail
+    - One idea is concatenate in URL as query string parameter and adapt reposync extract and use it
+
+Url to be used should be the one pointing to the cloud RMT server, saved in previous step, instead the one defined in table `susesccrepository`.
```

Comment:
In case of multiple options, what should be prefered? I think it also depends on where Uyuni Server is running. If it is running in the cloud, the local RMT servers should be prefered, while if the Server is running outside of the Cloud, SCC should be prefered. This detail could be seen as 2nd iteration step.

Reply:
Added it to the unresolved questions